src/mapdata.h

```diff
@@ -196,6 +209,7 @@ struct map_data_common_t {
 
     map_bash_info        bash;
     map_deconstruct_info deconstruct;
+    map_dismantle_info   dismantle;
```

Comment:
Why have two separate blocks?
Why not just have a flag allow deconstructing without tools?